ReactWindows/ReactNative/Bridge/Queue/LayoutActionQueue.cs

```diff
@@ -8,12 +10,37 @@ namespace ReactNative.Bridge.Queue
     class LayoutActionQueue : DispatcherActionQueue
     {
 #if CREATE_LAYOUT_THREAD
-        private static readonly CoreApplicationView s_layoutApplicationView = CoreApplication.CreateNewView();
+        private const int E_ILLEGAL_METHOD_CALL = -2147483634; // 0x80000000E
 
+        private static readonly Lazy<CoreApplicationView> s_layoutApplicationView = new Lazy<CoreApplicationView>(() =>
```

Comment:
>s_layoutApplicationView [](start = 58, length = 23)

Will we have an opportunity to create another dispatcher thread at a later point? I worry about perf in the fallback case.

Reply:
Something can be thought of, but honestly I couldn't find a trivial (and provable correct) way to "hand over" the layout management items to the newly created view. In other words, I don't want to end up having the callback for item X (using main dispatcher view) racing with callback for item Y (using layout manager view).

The current mitigation (at least for the My People scenario) is that there's a narrow set of scenarios that can keep you in this state:
- if app not running and you start My People, you are in this constrained case.
- if you execute "launch main app" from within My People, we "protocol activate" the app (there's no other way to "let's launch main window" as far as I know, maybe I'm wrong). Shell creates a new process that replaces the one that hosted my people panel! So we're good here.
- if you launch the app directly when My People is opened, same process is used. This would be the opportunity (and the only need) to reverse this fallback. We don't have that now, so we remained constrained.
- if app was already running and you open my people, nothing special happens, we're good.

My proposal is to wait a little and see how other activations play along (calling when app not running, all those), and maybe I have an epiphany and can do something that doesn't look like a kludge.


---
In reply to: [166566424](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166566424) [](ancestors = 166566424)